### PR TITLE
[docker] Fixes #1659 Create docker in docker image based on skycoindev-cli

### LINKED ISSUE #1659: Docker in Docker dev image
... similar to [skycoin/skycoindev-cli](https://hub.docker.com/r/skycoin/skycoindev-cli) suitable for devops engineers working on Docker .

### PR DESCRIPTION
**Pull request by stdevNorge**
Fixes #1659 

Changes:
- New docker image including [docker in docker](https://hub.docker.com/_/docker/) for development 

Does this change need to mentioned in CHANGELOG.md?
Yes

### CODE REVIEW

File: docker/images/dev-docker/README.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
+# Supported tags and respective `Dockerfile` links
```
Comment (olemis) [inline docker/images/dev-docker/README.md]: Include Docker in Docker specifics in README file.

**Comment by olemis** [comment]
I do not know why the Docker CI check is blocking the merge . Build reports success .

File: docker/images/dev-docker/README.md [hunk 2]
```diff
@@ -2,22 +2,22 @@
...
 
 # How to use this image
 
 ## Initialize your development environment.
 
 ```sh
 $ mkdir src
-$ docker run --rm \
-    -v src:/go/src skycoin/skycoindev-cli \
+$ docker run --privileged --rm \
```
Comment (olemis) [inline docker/images/dev-docker/README.md]: @stdevNorge I am not sure of whether `--privileged` flag is needed for commands not related to `dind` . Could you please try without it ? If Skycoin commands work , then revert changes like this one.

Comment (stdevNorge) [reply docker/images/dev-docker/README.md]: That is correct, <code>--privileged</code> flag is not needed for commands not related to <code>dind</code> . I fix the README right now.

**Comment by olemis** [comment]
@gz-c Docker CI step works now .

File: docker/images/dev-docker/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+RUN set -ex; \
+    apt-get update; \
+    apt-get install -y --no-install-recommends \
+    btrfs-progs \
+    e2fsprogs \
+    iptables \
+    xfsprogs \
+    apt-transport-https \
+    ca-certificates \
+    curl \
```
Comment (stdevMauricio) [inline docker/images/dev-docker/Dockerfile]: `curl` is already installed on the base image. Please check if there are other packages of this list in the same situation.

File: docker/images/dev-docker/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    gnupg2 \
+    software-properties-common \
+    pigz ;\
+    apt-get clean
+
+RUN curl -fsSL https://download.docker.com/linux/debian/gpg | sudo apt-key add -
```
Comment (stdevMauricio) [inline docker/images/dev-docker/Dockerfile]: You should do this before the first apt-get update and apt-get install.

File: docker/images/dev-docker/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+RUN apt-key fingerprint 0EBFCD88
+RUN add-apt-repository \
+   "deb [arch=amd64] https://download.docker.com/linux/debian \
+   $(lsb_release -cs) \
+   stable"
+
+RUN set -ex; \
```
Comment (stdevMauricio) [inline docker/images/dev-docker/Dockerfile]: Move this packages to the first apt-get install. Use just one RUN to install every package. Less layers.

File: docker/images/dev-docker/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+    apt-get update; \
+    apt-get install -y --no-install-recommends \
+    docker-ce ;\
+    apt-get clean
```
Comment (stdevMauricio) [inline docker/images/dev-docker/Dockerfile]: After apt-get clean add `rm -rf /var/lib/apt/lists/*` It removes the packages cache, making the image smaller.

**Pull request merged**